cli/src/package.rs

```diff
@@ -1,36 +1,144 @@
-use std::{env::current_dir, fs, path::PathBuf};
-
+use crate::error::CliError;
 use directories::ProjectDirs;
+use futures::future::join_all;
+use modmark_core::Resolve;
+use serde_json;
+use std::{
+    env::current_dir,
+    fs::{self, create_dir_all, File},
+    io::copy,
+    path::PathBuf,
+};
 
-pub trait Resolve {
-    fn resolve(&self, path: &str) -> Option<Vec<u8>>;
-}
+pub struct PackageManager {}
 
-pub struct PackageManager;
 impl Resolve for PackageManager {
-    fn resolve(&self, path: &str) -> Option<Vec<u8>> {
-        let mut local_path = match current_dir() {
-            Ok(p) => p,
-            Err(_) => return None,
-        };
+    type Error = CliError;
+    fn resolve(&self, path: &str) -> Result<Vec<u8>, Self::Error> {
+        let out = tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap()
+            .block_on(resolve_package(path))?;
+
+        return Ok(out);
+    }
+    fn resolve_all(&self, paths: Vec<&str>) -> Result<Vec<Vec<u8>>, CliError> {
+        let out = tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap()
+            .block_on(resolve_packages(paths))?;
+
+        return Ok(out);
+    }
+}
+
+async fn resolve_packages(paths: Vec<&str>) -> Result<Vec<Vec<u8>>, CliError> {
+    let futures = paths.iter().map(|path| resolve_package(path));
+
+    let results = join_all(futures).await;
```

Comment:
I think this could be `join_all(futures).await.collect::<Vec<Result<Vec<u8>, CliError>>().into()` and return immediately.